tests/test_adaptor.hh

```diff
@@ -276,6 +279,140 @@ namespace rascal {
     json factory_args{};
   };
 
+  struct MultipleStructureManagerNLFixtureCenterMask {
+    using ManagerTypeHolder_t =
+        StructureManagerTypeHolder<StructureManagerCenters,
+                                   AdaptorNeighbourList>;
+    MultipleStructureManagerNLFixtureCenterMask() {
+      // random-number engine used
+      std::mt19937_64 rng{1242484542};
+
+      for (auto && filename : this->filenames) {
+        AtomicStructure<3> atomic_structure{};
+        atomic_structure.set_structure(filename);
+        auto n_atoms{atomic_structure.get_number_of_atoms()};
+        std::uniform_int_distribution<int> uni(1, n_atoms - 2);
+        for (auto && cutoff : this->cutoffs) {
+          for (auto && skin : this->skins) {
+            for (auto && consider_ghost_neighbours :
+                 this->consider_ghost_neighbours_list) {
+              atomic_structure.set_structure(filename);
+
+              json parameters;
+              json structure = atomic_structure;
+              json adaptors;
+              json ad1{
+                  {"name", "AdaptorNeighbourList"},
+                  {"initialization_arguments",
+                   {{"cutoff", cutoff},
+                    {"skin", skin},
+                    {"consider_ghost_neighbours", consider_ghost_neighbours}}}};
+              adaptors.push_back(ad1);
+
+              parameters["structure"] = structure;
+              parameters["adaptors"] = adaptors;
+
+              this->factory_args.emplace_back(parameters);
+
+              // generate a new structure
+              auto n_flips = uni(rng);
+              for (int i_it{0}; i_it < n_flips; ++i_it) {
+                auto i_idx = uni(rng);
+                atomic_structure.center_atoms_mask(i_idx) = false;
+              }
+
+              json structure_no_center = atomic_structure;
+              parameters["structure"] = structure_no_center;
+              this->factory_args.emplace_back(parameters);
+            }
+          }
+        }
+      }
+    }
+
+    ~MultipleStructureManagerNLFixtureCenterMask() = default;
+
+    const std::vector<bool> consider_ghost_neighbours_list{{false, true}};
+    const std::vector<std::string> filenames{
+        "reference_data/CaCrP2O7_mvc-11955_symmetrized.json",
+        "reference_data/molecular_crystal.json",
+        "reference_data/simple_cubic_8.json",
+        "reference_data/small_molecule.json"};
+    // const std::vector<std::string> filenames{
+    //     "reference_data/CaCrP2O7_mvc-11955_symmetrized.json"};
+    const std::vector<double> cutoffs{{2., 3.}};
+    const std::vector<double> skins{{0., 0.3}};
+
+    json factory_args{};
+  };
+
+  struct MultipleStructureManagerNLStrictFixtureCenterMask {
```

Comment:
These names are getting impossible to decipher; for a while, I thought this was a duplicate of the class above.

Reply:
Well... I can only agree with you